### ISSUE TITLE
Tabs are sometimes condensed into one group at browser launch.

### ISSUE THREAD

**Issue by Shrinra** [Rapptz/power-tabs#16]
Hello! Sorry to bother you with another bug. 

I've noticed that about 20% of the time, Power Tabs will completely mess up the tab sortings between the groups that I have at browser relaunch. For example, I have about 50 tabs spread about between four groups (let's call them Group1, Group2, Group3, Group4). When this issue occurs, all of the tabs from Group2, Group3, and Group4 will be removed from those groups and condensed into Group1. They will also not be grouped together in Group1, but rather completely interspersed between one another. You then have to manually resort them into proper groups once again. 

I'm not sure if anyone else has had this issue, but I figured I would report it just in case.

I am on macOS Sierra and currently running Firefox 60 from the Beta channel.

**Comment by Rapptz** [comment]
Yeah. This seems to be a bug people are experiencing (see #11) but I can't reproduce it so it's kind of hard to fix.

**Comment by Shrinra** [comment]
Whoops! Sorry, I totally missed that. This type of thing actually seems to be somewhat common with vertical tab WebExtensions that do more than just display tabs in the sidebar. I used Tree Style Tab WE for awhile, and my tree hierarchies would flatten 50% of the time on browser relaunch. I wonder if it might be a Firefox bug.

**Comment by Rapptz** [comment]
I'm just using the standard session storage APIs but I wonder if they get cleared for some reason.

**Comment by chripo** [comment]
running 58.0.1
after start all tabs in one group.

**Comment by malesa** [comment]
I have the same error (version 1.8.0) and i can easily reproduce it. It happens when FF crashes or is killed. After restoring the session all tabs are in one group.

**Comment by trancephorm** [comment]
Happens all the time here. I'm crying now as this is the only decent tab manager for current FF versions. Please developer, just please make it work. If you do, I would gladly give the donation!

**Comment by desrod** [comment]
> but I can't reproduce it so it's kind of hard to fix.

1. Install Quantum ESR 60
2. Launch with --no-remote --ProfileManager
3. Add some tabs
4. Install PowerTab
5. Move tabs into groups
6. Close browser and restart
7. Tabs are all now in the last-used tab group, not sorted into their groups

100% reproducible on any clean macOS or Linux machine.

**Comment by sergeevabc** [comment]
@Rapptz, have you tried the steps which were mentioned by @desrod?
